packages/quantic/cypress/integration/facets/timeframe-facet/timeframe-facet.cypress.ts

```diff
@@ -0,0 +1,365 @@
+import {configure} from '../../../page-objects/configurator';
+import {
+  InterceptAliases,
+  interceptSearch,
+  routeMatchers,
+} from '../../../page-objects/search';
+import {scope} from '../../../reporters/detailed-collector';
+
+import {TimeframeFacetExpectations as Expect} from './timeframe-facet-expectations';
+import {TimeframeFacetActions as Actions} from './timeframe-facet-actions';
+
+interface TimeframeFacetOptions {
+  field: string;
+  label: string;
+  withDatePicker: boolean;
+  isCollapsed: boolean;
+}
+
+describe('quantic-timeframe-facet', () => {
+  const pageUrl = 's/quantic-timeframe-facet';
+
+  function visitTimeframeFacet(
+    options: Partial<TimeframeFacetOptions>,
+    waitForSearch = true
+  ) {
+    interceptSearch();
+    cy.visit(pageUrl);
+
+    configure(options);
+
+    if (waitForSearch) {
+      cy.wait(InterceptAliases.Search);
+    }
+  }
+
+  function loadFromUrlHash(
+    options: Partial<TimeframeFacetOptions> = {},
+    urlHash: string
+  ) {
+    interceptSearch();
+    cy.visit(`${pageUrl}#${urlHash}`);
+    configure(options);
+  }
+
+  it('should show placeholder before search completes', () => {
+    visitTimeframeFacet({}, false);
```

Comment:
I feel like there's a race condition here for the expectation to run before the request completes. Do you prefer this over using the `interceptIndefinitely` function?

Reply:
So far the test never failed on my end, but you have a very good point as it could be flaky when running in the CI. I don't mind using `interceptIndefinitely` instead.